File: StarterBots/JSBot/src/models/botStateDto.js
class Location {
    constructor(x, y) {
        this.x = x;
        this.y = y;
    }
}

class PowerUpLocation {
    constructor(location, type) {
        this.location = location;
        this.type = type;
    }
}

class BotStateDto {
    constructor(x, y, connectionId, elapsedTime, gameTick, heroWindow, directionState, leaderBoard, botPositions, powerUpLocations, weeds, powerUp = 0, superPowerUp = 0) {
        this.x = x;
        this.y = y;
        this.connectionId = connectionId;
        this.elapsedTime = elapsedTime;
        this.gameTick = gameTick;
        this.heroWindow = heroWindow;
        this.directionState = directionState;
        this.leaderBoard = leaderBoard;
        this.botPositions = botPositions;
        this.powerUpLocations = powerUpLocations;
        this.weeds = weeds;
        this.powerUp = powerUp;
        this.superPowerUp = superPowerUp;
    }
}

File: StarterBots/Kotlin/src/main/kotlin/models/dtos/BotStateDto.kt
package models.dtos

data class Location(var x: Int, var y: Int)
data class PowerUpLocation(var location: Location, var type: Int)


data class BotStateDto(var x: Int,
                  var y: Int,
                  var connectionId: String,
                  var elapsedTime: String,
                  var gameTick: Int,
                  var heroWindow: Array<IntArray>,
                  var directionState: Int,
                  var leaderBoard: Map<String, Int>,
                  var botPositions:Array<Location>,
                  var powerUpLocations: Array<PowerUpLocation>,
                  var weeds: Array<BooleanArray>,
                  var powerUp: Int = 0,
                  var superPowerUp: Int = 0

) {

    override fun toString(): String {
        val windowStr = StringBuilder()
        if (heroWindow != null) {
            for (y in heroWindow!![0].indices.reversed()) {
                for (x in heroWindow!!.indices) {
                    windowStr.append(heroWindow!![x][y])
                }
                windowStr.append("\n")
            }
        }
       return String.format("Position: (%d, %d), Direction: %s, GameTick: %d\n%s", x, y, directionState, gameTick, windowStr.toString())
    }
}
